src/KMonad/Model/Button.hs

```diff
@@ -81,6 +82,7 @@ import qualified RIO.HashSet as S
 data Button = Button
   { _pressAction   :: !Action -- ^ Action to take when pressed
   , _releaseAction :: !Action -- ^ Action to take when released
+  , _tapAction :: !Action -- ^ Action to take when tapped
```

Comment:
This honestly feels like a wonky change to me, as what counts as a "tap" or not is not really related to the button, but more to surrounding environment (even things like, say, key repeat)

Reply:
The problem is with buttons which listen for their own release, since they listen to the keycode not the Button.
`tap` in this context referrs to the `tap` function I.e. when you want to press and immideatly release a Button.
The perfect solution would be replacing the detecting of their own release to use the button state. This is a bit difficult since we mix a few things in the affected buttons. If we change the entire hook system to work with buttons instead we have the following problems:
1. ```closure
(defsrc a b)
(deflayer (around-next sft) #(x y))
```
    with `Pa Tb Ra` previously resulting in `Psft Tx Ty Rsft`
    and then resulting in `Psft Tx Rsft Ty` (is this wanted? probably not)
2. `#((around-next sft) a)` binded on `x` and the rest being fallthrough:
    `Tx Ty` priviously `Ta Psft Ty Rsft` and with the change `Psft Ta Rsft Ty` (both are kind of wierd)
3. `layer-next` requires the old system, since want to change the layer prior to button look up.

My point being:
It is not really defined how most buttons should behave in such context. Though something like nested `tap-macro`s have an intended behaviour, it is just broken, since the last key is still held.
This could happen when you have many `tap-macro`s and wanted to reduce work with aliases.
The different `implicit-around` buttons are the prime example for this going wrong. Changing them inside nested contexts is also not the correct solution. Cosider:
`#(A B)` bound on `x` with `fallthrough = true`:
`Px Ty Rx` results in the same problem `implicit-around`s tried to fix: `Psft Ta Rsft Psft Pb Ty Rsft` also written as `A B Y`. since the last button is held.

The affected buttons all have a `press` action but no `release` action. So other buttons who call `tap` priviously `press *> release` now use a third action which defaults to `press *> release` but it can be changed.

T.l.d.r.: I should change the comment

Reply:
Hey @slotThe, I was wondering if there was anything else or if this is ready to merge.
I really don't want to stall #879 as I myself requested a release in #857 two months ago.
Though I also like diligence in code review and respect work and free time.
(Just a ping to assure (mostly to myself) that this is not lost in emails).

Reply:
@jokesper sorry, I've been traveling for work and haven't had much time for this lately. I'll try to get to everything this week.

I still feel like this is more of a bandaid-change, but that's still better than nothing until we properly refactor the keycode handling :)